src/utils/core.js

```diff
@@ -132,3 +132,23 @@ export function calculateDimensions(arr) {
     }
     return dimensions;
 }
+
+/**
+ * Replicate python's .pop() method for objects.
+ * @param {Object} obj The object to pop from.
+ * @param {string} key The key to pop.
+ * @param {*} defaultValue The default value to return if the key does not exist.
+ * @returns {*} The value of the popped key.
+ * @throws {Error} If the key does not exist and no default value is provided.
+ */
```

Comment:
If you decide to use `"strictNullChecks": true`, then this will return the expected types:

![image](https://github.com/xenova/transformers.js/assets/5236548/4a5b2aa3-2224-4aae-a236-40dbae1eaeb6)


```suggestion
/**
 * Replicate python's .pop() method for objects.
 * @param {OBJ} obj The object to pop from.
 * @param {KEY} key The key to pop.
 * @param {DEF} [defaultValue=undefined] The default value to return if the key does not exist.
 * @template {Record<string, any>} OBJ
 * @template {string} KEY
 * @template DEF
 * @throws {Error} If the key does not exist and no default value is provided.
 * @returns {Exclude<OBJ[KEY] | DEF, undefined>} The value of the popped key.
 */
```